### PR TITLE
MF-888 - Add group, certs and bootstrap SDK tests

### LINKED ISSUE #888: Update SDK tests
Add missing SDK tests

### PR DESCRIPTION
**Pull request by zzokki81**
### What does this do?
Added SDK tests
### Which issue(s) does this PR fix/relate to?
Resolves #888 

### Have you included tests for your changes?
SDK Tests
### Did you document any new/modified functionality?
No

### CODE REVIEW

File: pkg/sdk/go/bootstrap_test.go [hunk 1]
```diff
@@ -234,7 +228,7 @@ func TestViewBootstrap(t *testing.T) {
 			err:  createError(sdk.ErrFailedFetch, http.StatusNotFound),
 		},
 		{
-			desc: "view a config with invalid token",
+			desc: "view a config with wrong credentials",
```
Comment (manuio) [inline pkg/sdk/go/bootstrap_test.go]: `invalid token` is probably clearer

File: pkg/sdk/go/bootstrap_test.go [hunk 2]
```diff
@@ -287,17 +280,17 @@ func TestUpdateBootstrap(t *testing.T) {
 	cases := []struct {
 		desc   string
 		auth   string
-		config mfsdk.BootstrapConfig
+		config sdk.BootstrapConfig
 		err    error
 	}{
 		{
-			desc:   "update with invalid token",
+			desc:   "update config with wrong credentials",
```
Comment (manuio) [inline pkg/sdk/go/bootstrap_test.go]: I'd keep `update config with invalid token`

File: pkg/sdk/go/bootstrap_test.go [hunk 3]
```diff
@@ -287,17 +280,17 @@ func TestUpdateBootstrap(t *testing.T) {
...
 			auth:   invalidToken,
 			config: updatedConfig,
 			err:    createError(sdk.ErrFailedUpdate, http.StatusUnauthorized),
 		},
 		{
-			desc:   "update with empty token",
+			desc:   "update config  with empty token",
```
Comment (manuio) [inline pkg/sdk/go/bootstrap_test.go]: There are 2 white spaces after `config`

File: pkg/sdk/go/bootstrap_test.go [hunk 4]
```diff
@@ -365,7 +356,7 @@ func TestUpdateBootstrapCerts(t *testing.T) {
 			err:        createError(sdk.ErrFailedCertUpdate, http.StatusNotFound),
 		},
 		{
-			desc:       "update cert with invalid token",
+			desc:       "update cert with with wrong credentials",
```
Comment (manuio) [inline pkg/sdk/go/bootstrap_test.go]: Same here

File: pkg/sdk/go/bootstrap_test.go [hunk 5]
```diff
@@ -423,13 +414,13 @@ func TestRemoveBootstrap(t *testing.T) {
 		err   error
 	}{
 		{
-			desc:  "remove with invalid token",
+			desc:  "remove config with wrong credentials",
```
Comment (manuio) [inline pkg/sdk/go/bootstrap_test.go]: and here

**Review by manuio — CHANGES_REQUESTED** [review]

File: pkg/sdk/go/certs.go [hunk 6]
```diff
@@ -96,4 +83,5 @@ type certReq struct {
 	KeyType    string `json:"key_type"`
 	Encryption string `json:"encryption"`
 	Valid      string `json:"valid"`
```
Comment (manuio) [inline pkg/sdk/go/certs.go]: You can remove `Valid` field since it's replaced by `TTL`

File: pkg/sdk/go/certs_test.go [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+		keyBits int
+		keyType string
+		ttl     string
+		token   string
+		err     error
+	}{
+		{
+			desc:    "issue new cert with invalid token",
+			thingID: thingID,
+			keyBits: 2048,
```
Comment (manuio) [inline pkg/sdk/go/certs_test.go]: Create a const and use it in all cases

File: pkg/sdk/go/responses.go [hunk 8]
```diff
@@ -22,7 +22,7 @@ type createChannelsRes struct {
 	Channels []Channel `json:"channels"`
 }
 
-type pageRes struct {
+type PageRes struct {
```
Comment (manuio) [inline pkg/sdk/go/responses.go]: Is this really needed?

**Review by manuio — CHANGES_REQUESTED** [review]

File: pkg/sdk/go/keys_test.go [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+			err:      createError(sdk.ErrFailedCreation, http.StatusUnauthorized),
+		},
+		{
+			desc:     "issue key with no duration",
+			token:    token,
+			duration: 0,
+			err:      nil,
+		},
+		{
+			desc:     "Issue a new key",
```
Comment (manuio) [inline pkg/sdk/go/keys_test.go:54]: `issue` would be more consistent here

File: pkg/sdk/go/certs_test.go [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+			desc:    "issue new cert for non-existing thing",
+			thingID: wrongID,
+			keyBits: keyBits,
+			keyType: key,
+			ttl:     ttl,
+			token:   token,
+			err:     createError(sdk.ErrFailedCreation, http.StatusInternalServerError),
+		},
+		{
+			desc:    "issue new cert with no thing id",
```
Comment (manuio) [inline pkg/sdk/go/certs_test.go]: `emtpy thing ID`

**Review by manuio — CHANGES_REQUESTED** [review]

**Comment by dborovcanin** [comment]
@zzokki81 Can you, please, resolve conflicts so we can review the PR?

**Comment by zzokki81** [comment]
> @zzokki81 Can you, please, resolve conflicts so we can review the PR?
Sure, I'll take a look at that.

**Review by arvindh123 — DISMISSED** [review]

**Comment by dborovcanin** [comment]
This PR has some interesting ideas, but it's been out of date with no signs of any updates, so I'm closing it.

**Pull request closed without merging**